### DISCUSSION TITLE
Possible features

### DISCUSSION THREAD

**Discussion by 2118557404** [winstxnhdw/lc-hax#235] [Ideas]
There is a hidden instruction to resign everyone in the terminal, which should become a new feature.

**Comment by winstxnhdw** [comment]
What secret command?

**Reply by 2118557404** [reply]
Enter 'eject' in the terminal

**Reply by winstxnhdw** [reply]
LOL ok i'll add it

**Reply by o777l** [reply]
😃

Be a first!

**Comment by winstxnhdw** [comment]
Seems like you can only use it as host :(

**Reply by 2118557404** [reply]
On.That's not very entertaining, but it could be a useful one for hosts, a little-known one)

**Reply by winstxnhdw** [reply]
i might just make a /terminal command for all terminal commands honestly

**Reply by GigglingMinister** [reply]
that'd be convenient

**Comment by 2118557404** [comment]
Oh, there is another problem. If you provoke a monster after it dies, it will still be commanded but will not attack, instead it will make a sound. It is unclear if this is related to the command to respawn monsters. If you can prevent the monster from being provoked after it dies, or if it must be resurrected before it can be provoked, it will be more interesting.

**Reply by winstxnhdw** [reply]
That only happens if you use funny revive.

**Reply by 2118557404** [reply]
I didn't use m3+r to resurrect monsters. When they were killed by friends, I used the anger command again, but they still made sounds.

**Reply by winstxnhdw** [reply]
I think only the centipede does that.

**Reply by 2118557404** [reply]
I remember some impressions that are more vivid. Half-bodied fish can cry and spiders can spin silk. You are right, not all monsters can do that.

**Reply by winstxnhdw** [reply]
Yeap, you're right. Thumper and spider should be able to do that too. They all have one thing in common: I am calling their RPC commands directly.

**Reply by 2118557404** [reply]
It would be much easier to add an instruction to delete specific monsters, which would allow us to keep these interesting ones or delete them.)

**Reply by winstxnhdw** [reply]
You can now do this with /hitclick

**Reply by 2118557404** [reply]
That's great. Thanks. Best hack I've ever used

**Comment by 2118557404** [comment]
What seems to irritate the Nutcracker in the game is the gun in his hand, which mimics the firing of the nutcracker when he is shot. In the latest version, if you use the free camera '=' after being killed, the game screen will get stuck and cannot be recovered. Typing /hitclick indicates that the command was not found. You can add a trigger to the medium key to make the gun go off.

**Reply by winstxnhdw** [reply]
@armorfid looks like we forgot to register the `hitclick` command 😂

**Reply by winstxnhdw** [reply]
> What seems to irritate the Nutcracker in the game is the gun in his hand, which mimics the firing of the nutcracker when he is shot.

What does this mean?

**Reply by 2118557404** [reply]
The object of command anger is the Nutcracker's gun, whose property is to attack the entire map, firing a bullet each time it provokes, and mimicking the attack in the Nutcracker's hands, regardless of whether the Nutcracker sees the enemy or how far away the enemy is.

**Reply by winstxnhdw** [reply]
How do you know this?

**Reply by 2118557404** [reply]
When my teammate holds the Nutcracker gun and uses the command to provoke the monster to the other teammates, the nutcracker gun in his hand will fire itself and it will be the nutcracker firing animation, and when he puts the gun away, it will be the nutcracker firing animation, and the reload will be the nutcracker firing animation, making the gun infinitely loaded.

**Reply by 2118557404** [reply]
The gun that went through the nutcracker's hand would have provoked anyone, anywhere

**Reply by winstxnhdw** [reply]
How did you managed to grab the gun from the nutcracker while it's still alive?

**Reply by armorfid** [reply]
> @armorfid looks like we forgot to register the `hitclick` command 😂

Didn't `class Chat` remove the need to manually register commands? I didn't have to register it, worked right off the bat. I think @2118557404 simply hasn't pulled latest version.

**Reply by winstxnhdw** [reply]
Yeap but we still have to 'register' it like so.

```cs
[Command("/stunclick")]
public class StunOnClickCommand : ICommand {
    public void Execute(string[] _) {
        Setting.EnableStunOnLeftClick = !Setting.EnableStunOnLeftClick;
        Chat.Print($"Stunclick: {(Setting.EnableStunOnLeftClick ? "Enabled" : "Disabled")}");
    }
}
```

I looked at the PR and yes, it seems we indeed did not add a new command.

**Reply by armorfid** [reply]
oops I forgor to `git add` the file 💀

**Comment by 2118557404** [comment]
Hello, are you planning to strengthen the anti-kick out function? Recently, a client blacklist ID module has appeared. Once you are listed on this blacklist, you will be notified and kicked out directly in the lobby. It is more efficient than the in-game kick out function (calling the SteamID library). It is now on the cloud module, and many rooms are using it.

**Reply by winstxnhdw** [reply]
Honestly, I am out of ideas on how to deal with this right now.

**Reply by armorfid** [reply]
The anti-kick is just a silly exploit made possible due to some bad code in the vanilla game. Third-party mods are better built, in ways that are usually exploit-proof. Not much that can be done.

**Reply by 2118557404** [reply]
[小海黑名单MOD2.1.zip](https://github.com/winstxnhdw/lc-hax/files/13941250/MOD2.1.zip)
This is his runtime library, but there is no source code because he definitely won't release it. Therefore, all existing mods or cheats are now limited by its influence.（

**Reply by winstxnhdw** [reply]
Well, it's nothing really special. He's just adding the player's Steam ID to a dictionary.

```cs
this.backstr = plugin.Instance.blackListConfig.AddBlackList(component.playerSteamId.ToString(), component.playerUsername);
```

**Reply by 2118557404** [reply]
Yes, the production of this game is much simpler, and it can always be changed.

**Reply by 2118557404** [reply]
I thought the server id was called in the lobby.

**Reply by 2118557404** [reply]
If he adds his Steam ID to the lobby blacklist and is replaced by another ID, this will solve the problem and prevent kickouts at any time.

**Reply by winstxnhdw** [reply]
The problem is changing your Steam ID seems to break the game.
